apis/metal3.io/v1alpha1/firmwareschema_types.go

```diff
@@ -80,56 +82,65 @@ type FirmwareSchema struct {
 }
 
 // Check whether the setting's name and value is valid using the schema
-func (host *FirmwareSchema) CheckSettingIsValid(name string, value intstr.IntOrString, schemas map[string]SettingSchema) bool {
+func (host *FirmwareSchema) CheckSettingIsValid(name string, value intstr.IntOrString, schemas map[string]SettingSchema) error {
 
 	schema, ok := schemas[name]
 	if !ok {
-		// The setting must exist in the status
-		return false
+		return fmt.Errorf("Setting %s is not in the schema", name)
 	}
 
 	if schema.ReadOnly != nil && *schema.ReadOnly == true {
-		return false
+		return fmt.Errorf("Setting %s is ReadOnly", name)
 	}
 
 	// Check if valid based on type
 	switch schema.AttributeType {
 	case "Enumeration":
 		for _, av := range schema.AllowableValues {
 			if value.String() == av {
-				return true
+				return nil
 			}
 		}
-		return false
+		return fmt.Errorf("Setting %s uses invalid enum %s", name, value.String())
 
 	case "Integer":
 		if schema.LowerBound == nil || schema.UpperBound == nil {
 			// return true if no settings to check validity
-			return true
+			return nil
+		}
+		if value.IntValue() >= *schema.LowerBound && value.IntValue() <= *schema.UpperBound {
+			return nil
 		}
-		return (value.IntValue() >= *schema.LowerBound && value.IntValue() <= *schema.UpperBound)
+		return fmt.Errorf("Setting %s integer %s is out of range, %d - %d", name, value.String(), *schema.LowerBound, *schema.UpperBound)
 
 	case "String":
 		if schema.MinLength == nil || schema.MaxLength == nil {
```

Comment:
Same, in theory nothing prevents us from having only upper limit.

Reply:
Yep, changed this to evaluate them separately.